src/ext/ws.js

```diff
@@ -151,6 +146,21 @@ This extension adds support for WebSockets to htmx.  See /www/extensions/ws.md f
 
 			api.settleImmediately(settleInfo.tasks);
 			api.triggerEvent(socketElt, "htmx:wsAfterMessage", { message: response, socketWrapper: socketWrapper.publicInterface })
+		}
+
+		socketWrapper.addEventListener('message', function (event) {
+			if (maybeCloseWebSocketSource(socketElt)) {
+				return;
+			}
+
+			var response = event.data;
+			if (socketWrapper.socket.binaryType === "blob") {
+				response.text()
+					.then(responseHandler);
+			} else { // arraybuffer
+				var responseStr = String.fromCharCode.apply(null, new Uint16Array(response));
+				responseHandler(responseStr);
+			}
```

Comment:
This change breaks support for `text` message type, which still can be `string`, regardless of socket's binary type (https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/data#value). I checked the test server in `test/realitime/server.go`, it doesn't work.

Reply:
It looks like everything is working with the latest changes.